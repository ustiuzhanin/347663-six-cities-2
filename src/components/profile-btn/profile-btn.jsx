import React from "react";
import PropTypes from "prop-types";

import { Link } from "react-router-dom";
import { connect } from "react-redux";

/* eslint-disable camelcase*/

const ProfileBtn = (props) => {
  const { isAuthorizationRequired, user } = props;
  return (
    <nav className="header__nav">
      <ul className="header__nav-list">
        <li className="header__nav-item user">
          <Link
            to={isAuthorizationRequired ? "/login" : "/profile"}
            className="header__nav-link header__nav-link--profile"
          >
            <div className="header__avatar-wrapper user__avatar-wrapper"></div>
            <span className="header__login">
              {isAuthorizationRequired ? "Sign up" : user.email}
            </span>
          </Link>
        </li>
      </ul>
    </nav>
  );
};

ProfileBtn.propTypes = {
  isAuthorizationRequired: PropTypes.bool.isRequired,
  user: PropTypes.shape({
    avatar_url: PropTypes.string,
    email: PropTypes.string,
    id: PropTypes.number,
    is_pro: PropTypes.string,
    name: PropTypes.string,
  }).isRequired,
};

const mapStateToProps = (state) =>
  Object.assign({}, null, {
    user: state.auth.user,
    isAuthorizationRequired: state.auth.isAuthorizationRequired,
  });
/* eslint-enable camelcase*/

export { ProfileBtn };

export default connect(mapStateToProps, null)(ProfileBtn);
