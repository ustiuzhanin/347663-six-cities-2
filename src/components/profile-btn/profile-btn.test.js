import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import { ProfileBtn } from "./profile-btn.jsx";

test(`ProfileBtn's Snapshot`, () => {
  const renderer = new ShallowRenderer();
  renderer.render(<ProfileBtn url="test.com" title="test" />);

  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
