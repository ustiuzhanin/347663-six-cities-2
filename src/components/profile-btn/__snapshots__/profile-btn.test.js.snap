// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileBtn's Snapshot 1`] = `
<li
  className="header__nav-item user"
>
  <ForwardRef
    className="header__nav-link header__nav-link--profile"
    to="test.com"
  >
    <span
      className="header__login"
    >
      test
    </span>
  </ForwardRef>
</li>
`;
